src/core/public/ui_settings/ui_settings_api.ts

```diff
@@ -65,16 +68,17 @@ export class UiSettingsApi {
    * already in progress it will wait until the previous request is complete
    * before sending the next request
    */
-  public batchSet(key: string, value: any) {
+  public batchSet(key: string, value: any, scope?: UiSettingScope) {
     return new Promise<UiSettingsApiResponse | undefined>((resolve, reject) => {
       const prev = this.pendingChanges || NOOP_CHANGES;
-
+      const prevValues = { ...prev.values };
+      const scopedKey = scope ?? 'undefined';
```

Comment:
Seems this `undefined` string will be shared in many places, can we shared with a constant variable?